task/08-objects-tasks.js

```diff
@@ -106,34 +110,105 @@ function fromJSON(proto, json) {
  *  For more examples see unit tests.
  */
 
+class Selector {
+    
+    result = [];
+    currentItems = [];
+    
+    stringify() {
+        return this.result;
+    }
+
+    element(value) {
+        this.checkForRepeat(1);
+        this.result += value;
+
+        return this;
+    }
+
+    id(value) {
+        this.checkForRepeat(2);
+        this.result += '#' + value;
+        
+        return this;
+    }
+
+    class(value) {
+        this.checkForRepeat(3);
+        this.result += '.' + value;
+        
+        return this;
+    }
+
+    attr(value) {
+        this.checkForRepeat(4);
+        this.result += '[' + value + ']';
+        
+        return this;
+    }
+
+    pseudoClass (value) {
+        this.checkForRepeat(5);
+        this.result += ':' + value;
+        
+        return this;
+    }
+
+    pseudoElement (value) {
+        this.checkForRepeat(6);
+        this.result += '::' + value;
+        
+        return this;
+    }
+
+    combine(selector1, combinator, selector2){
+        this.result = selector1.result + ' ' + combinator + ' ' + selector2.result;
+        
+        return this;
+    }
+
+    checkForRepeat(item) {
+        
+        if(this.currentItems.length > 0){
+
+            if(this.currentItems.some((value) => item < value)){
+                throw new Error("Selector parts should be arranged in the following order: element, id, class, attribute, pseudo-class, pseudo-element");
+                }
+            if( this.currentItems.some((value) => item == value) && (item == 1 || item == 2 || item == 6)){
```

Comment:
"operator === is always preferable
sometimes ''==' may cause an error
(check at console :
"0" == 0
and "0" === 0)"